core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/csv/CSVScanSourceOpExec.scala

```diff
@@ -6,43 +6,49 @@ import edu.uci.ics.texera.workflow.common.tuple.Tuple
 import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeTypeUtils, Schema}
 
 import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
+import scala.util.{Failure, Success, Try}
 
-class CSVScanSourceOpExec private[csv] (
-    val localPath: String,
-    val schema: Schema,
-    val customDelimiter: Char,
-    val hasHeader: Boolean
-) extends SourceOperatorExecutor {
-  var rows: Iterator[Seq[String]] = _
+class CSVScanSourceOpExec private[csv] (val desc: CSVScanSourceOpDesc)
+    extends SourceOperatorExecutor {
+  val schema: Schema = desc.inferSchema()
   var reader: CSVReader = _
-  override def produceTexeraTuple(): Iterator[Tuple] =
-    new Iterator[Tuple]() {
-      override def hasNext: Boolean = rows.hasNext
-
-      override def next: Tuple = {
-        // obtain String representation of each field
-        val fields: Seq[String] = rows.next
-        // parse Strings into inferred AttributeTypes
-        val parsedFields: Array[Object] = AttributeTypeUtils.parseFields(
-          fields.toArray,
-          schema.getAttributes
-            .map((attr: Attribute) => attr.getType)
-            .toArray
-        )
-        Tuple.newBuilder(schema).addSequentially(parsedFields).build
-      }
 
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+
+    // skip line if this worker reads the start of a file, and the file has a header line
+    val startOffset = desc.offset.getOrElse(0).asInstanceOf[Int] + (if (desc.hasHeader) 1 else 0)
+    var tuples = reader.iterator
+      .map(fields =>
+        Try({
+          val parsedFields: Array[Object] = AttributeTypeUtils.parseFields(
+            fields.toArray,
+            schema.getAttributes
+              .map((attr: Attribute) => attr.getType)
+              .toArray
+          )
+          Tuple.newBuilder(schema).addSequentially(parsedFields).build
+        }) match {
+          case Success(tuple) => tuple
+          case Failure(_)     => null
+        }
+      )
+      .drop(startOffset)
+
+    desc.limit match {
```

Comment:
my personal perference for these type of statements is to just use a `if () { ... }`. it's more like java and easier to understand.

Reply:
changed.